File: src/models/Inventory.ts
import mongoose, { Schema, Model } from 'mongoose';
import { IInventory } from '../types';

const inventorySchema = new Schema<IInventory>(
    {
        name: {
            type: String,
            required: true,
        },
        category: {
            type: String,
            required: true,
        },
        stock: {
            type: Number,
            required: true,
            min: 0,
        },
        minStock: {
            type: Number,
            required: true,
            min: 0,
        },
        unit: {
            type: String,
            required: true,
        },
        price: {
            type: Number,
            required: true,
            min: 0,
        },
    },
    {
        timestamps: true,
    }
);

// Create indexes
inventorySchema.index({ name: 1 });
inventorySchema.index({ category: 1 });

export const Inventory: Model<IInventory> = mongoose.model<IInventory>('Inventory', inventorySchema);

File: src/main/java/com/medicare/backend/models/Inventory.java
package com.medicare.backend.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "inventory")
public class Inventory {
    @Id
    private String id;

    @Indexed
    private String name;

    @Indexed
    private String category;

    private Integer stock;

    private Integer minStock;

    private String unit;

    private Double price;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}
